### DISCUSSION TITLE
EXTRA MIDI CHANNELS OVER USB [might need external device]

### DISCUSSION THREAD

**Discussion by worker-gray** [SynthstromAudible/DelugeFirmware#1015] [Ideas]
We have 16 midi channels-- we'll call that 'Block A'
I propose another 16 channels [Block B] and maybe a Block C to boot. It might require an external device to route the channels to different outputs.

**Comment by dominickolb** [comment]
I just wanted to reqest a feature and then saw that yours is very similar.

Wouldn't it be nice if we could cycle throug the MIDI Output (on a Miditrack but also in the Kits) between MIDI-DIN Out / USB-MIDI Out and BOTH-MIDI Out?
For me it would be very helpful

**Comment by sapphire-arches** [comment]
If you're on OSX you might need to reset the device configuration for the Deluge in "MIDI Audio Setup" -- the community firmware has 3 MIDI cables on USB, two of which can be used for general MIDI data (the third is reserved for only SYSEX processing).

**Reply by dominickolb** [reply]
Yes, but the "problem" is that a MIDI track on deluge always ouputs its signal to USB AND DIN-MIDI. For me it would be great to have a choice to only output the signal to USB OR DIN-MIDI (or both like now)

**Comment by worker-gray** [comment]
For clarity- I'm suggesting being able to control AT LEAST 32 separate instruments (2x16 channels) via MIDI over USB. This might understandably require an external device to split each block of 16 instrument channels into multiple DIN outputs.

**Comment by sapphire-arches** [comment]
Oh I see. Yeah, we should probably support that, at least differentiating between USB virtual cables and DIN out

**Comment by rhombuslabs** [comment]
I'd love to use a multi-port MIDI interface directly. I have a MOTU MIDI Express XT, I don't believe it's class compliant but it can run standalone and I'm going through DIN so I only have access to 16 channels anyways. If I could specify a port (not just DIN, USB, but also all output ports available on USB) in addition to channel then I could simplify and downsize at a minimum. 

I could daisy chain with just a 2 output interface for example. 

Or I was also looking at the MRCC. It can route USB-B and DIN MIDI devices which would be great for integrating my USB keyboard in a flexible manner to both the Deluge as well as directly to other hardware.
